### PR TITLE
Centralize Set-RdsContext calls into SetTenantGroupContextAndValidate

### PR DESCRIPTION
**Pull request by lintFan**

### CODE REVIEW

**Comment by lintFan** [comment]
We may want to look into why we see more detail when outputing error info using
$_ | Out-String
when running on Azure using DSC VS when I run functions locally (like AuthenticateRdsAccount) . It works but it is a bit verbose on Azure.

File: wvd-templates/DSC/Functions.ps1 [hunk 1]
```diff
@@ -273,41 +278,82 @@ function AuthenticateRdsAccount {
...
+                    if($innerExAsStr.Length -gt 0)
+                    {
+                        $innerExAsStr += "`n"
+                    } 
+                    $innerExAsStr += $innerEx
+                }
+            }
+            
+            $errMsg = "Error authenticating Windows Virtual Desktop account, ServicePrincipal=" + $ServicePrincipal
+            $errMsg += " Error Details:`n$($_ | Out-String)"
```
Comment (nakranimohit0) [inline wvd-templates/DSC/Functions.ps1]: I see that you are using `$PSItem | Out-String` in function `SetTenantGroupContextAndValidate` vs `$_ | Out-String` here. Whats the diff ?

**Comment by nakranimohit0** [comment]
Should I merge this branch (https://github.com/Azure/RDS-Templates/tree/checkRDPSModVersion/) to this PR to include changes to check the imported powershell module version ?

**Comment by lintFan** [comment]
The changes to the AuthenticateRdsAccount function was removed from this and is now in this PR #356

**Pull request merged**